### ISSUE TITLE
learning_rate adjustment for run_multi_process, run_single_process

### ISSUE THREAD

**Issue by iclementine** [clab/dynet#1527]
First of all, I really appreciate the clean design and abstraction of dynet/mp. When I read the code of `ILearner`, `SufficientStats`, `Datum`, `run_single_process` and `run_multi_process`, I finally find that it is time to stop writing tedious training function to loop over training dataset and report every N updates, test on dev dateset every epoch, and keep track of a loss accumulator or sth like that.

When using `run_single_process` or `run_multi_process`, just write a function for a single instance is enough. It is really cool.

But the problem is how to update the learning rate of the trainer when using these high-level abstraction. the `run_child` function does not include learning rate update. Is there a clean way to add code for that in the single level of abstraction?
